modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/AbstractResourceURLBuilder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.charon3.core.protocol.endpoints;
+
+import org.wso2.charon3.core.exceptions.NotFoundException;
+
+import java.util.Map;
+
+/**
+ * This is an abstract layer for performing operations/customizations for the resource endpoints exposed by the
+ * server.
+ */
+public class AbstractResourceURLBuilder {
```

Comment:
This is not exactly an abstract resource builder right.
Either we can have an interface and a default implementation in SCIM

Reply:
shall I follow the same pattern as https://github.com/wso2/carbon-identity-framework/blob/master/components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/ServiceURLBuilder.java#L29-L35 here?

Reply:
fixed.